durabletask-azuremanaged/durabletask/azuremanaged/client.py

```diff
@@ -27,14 +26,7 @@ def __init__(self, *,
         # Append DurableTask-specific metadata
         self._metadata.append(("taskhub", taskhub))
         self._metadata.append(("dts", "True"))
-        self._metadata.append(("use_managed_identity", str(use_managed_identity)))
-        self._metadata.append(("client_id", str(client_id or "None")))
-
-        self._access_token_manager = AccessTokenManager(use_managed_identity=use_managed_identity,
-                                                        client_id=client_id)
-        token = self._access_token_manager.get_access_token()
-        self._metadata.append(("authorization", token))
-
+        self._metadata.append(("token_credential", token_credential))

```

Comment:
This should not be added to the `_metadata` collection. "Metadata" is for key-value pairs that get added to the request, which this is not. Instead, pass it to the `DTSDefaultClientInterceptorImpl` constructor directly. On top of that, there should be no need for `_metadata` at all in this class, so let's just remove it.